tools.py

```diff
@@ -0,0 +1,29 @@
+import getpass
+import requests
+import urllib
+import urlparse
+
+
+requests.packages.urllib3.disable_warnings()
+
+def vsphere_url(vm, host, args):
+    # Generates console URL for vSphere versions prior 6.0
+    ticket = vm.AcquireTicket('mks')
+    vm_host = ticket.host if ticket.host else host
+    path = '?host={0}&port={1}&ticket={2}&cfgFile={3}&thumbprint={4}'.format(
+        vm_host, ticket.port, ticket.ticket, ticket.cfgFile,
+        ticket.sslThumbprint)
+    base_url = "http://rgerganov.github.io/noVNC/5"
```

Comment:
We don't use this in this code anyways, it's there for compatibility with the orig dev. So, no.